tee-worker/litentry/core/assertion-build/src/oneblock/mod.rs

```diff
@@ -0,0 +1,168 @@
+// Copyright 2020-2023 Litentry Technologies GmbH.
+// This file is part of Litentry.
+//
+// Litentry is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Litentry is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Litentry.  If not, see <https://www.gnu.org/licenses/>.
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+extern crate sgx_tstd as std;
+
+pub mod course;
+
+use crate::*;
+use http::header::{AUTHORIZATION, CONNECTION};
+use http_req::response::Headers;
+use itc_rest_client::{error::Error as RestClientError, RestGet, RestPath};
+use lc_data_providers::{build_client, GLOBAL_DATA_PROVIDER_CONFIG};
+use serde::{Deserialize, Serialize};
+use std::string::ToString;
+
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct OneBlockResponse {
+	#[serde(flatten)]
+	data: serde_json::Value,
+}
+
+impl RestPath<String> for OneBlockResponse {
+	fn get_path(path: String) -> core::result::Result<String, RestClientError> {
+		Ok(path)
+	}
+}
+
+fn fetch_data_from_notion(course_type: &OneBlockCourseType) -> Result<OneBlockResponse> {
+	let mut headers = Headers::new();
+	headers.insert(CONNECTION.as_str(), "close");
+	headers.insert("Notion-Version", "2022-06-28");
+	headers.insert(
+		AUTHORIZATION.as_str(),
+		GLOBAL_DATA_PROVIDER_CONFIG.read().unwrap().oneblock_notion_key.clone().as_str(),
+	);
+
+	let mut client = build_client(
+		GLOBAL_DATA_PROVIDER_CONFIG.read().unwrap().oneblock_notion_url.clone().as_str(),
+		headers,
+	);
+
+	client.get::<String, OneBlockResponse>(String::default()).map_err(|e| {
+		Error::RequestVCFailed(
+			Assertion::Oneblock(course_type.clone()),
+			ErrorDetail::DataProviderError(ErrorString::truncate_from(
+				format!("{e:?}").as_bytes().to_vec(),
+			)),
+		)
+	})
+}
+
+fn oneblock_course_result(
+	data: &serde_json::Value,
+	course_type: &OneBlockCourseType,
+	addresses: Vec<String>,
+) -> Result<bool> {
+	let get_results = |data: &serde_json::Value| -> Option<Vec<serde_json::Value>> {
+		data.get("results").and_then(|results| results.as_array()).cloned()
+	};
+
+	// A object contains 6 rows of content
+	let get_table_row = |object: &serde_json::Value| -> Option<Vec<serde_json::Value>> {
```

Comment:
Is this notion data structure ? If yes, can we have a separate module for parsing it ?

Reply:
Updated.